### ISSUE TITLE
[False-Positive] One-time use variables + annotations

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1113]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False Positive                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.9 |
| Language level | PHP 7.2                        |

Current behaviour:
---

First of all: Not absolutely sure if it's a false positive, but for me it is.

Please see the following example code:

```php
<?php

declare(strict_types=1);

abstract class IntegrationTestCaseAbstract extends TestCase
{
    protected function getFooById(int $fooId): FooInterface
    {
        /** @var FooService $fooService */
        $fooService = $this->getFromContainer(FooService::class);

        return $fooService->getById($fooId);
    }
}
```

The inspection "One-time use variables" here reports: "Variable $fooService is redundant.". Well yes, it is redundant, but if I inline the variable as suggested, PhpStorm does not recognise anymore that `getById()` is called on `FooService` and thus no refactoring, navigating with STRG, etc.pp. works anymore.

Suggested behaviour:
---

Do not report in this case, as the annotation is required for type hinting here, so that the IDE understands it. Basically, I'd suggest to not report one-time variables if they have an annotation.

**Comment by kalessil** [comment]
Has been fixed already, the fix will be in the next release.
